src/project.rs

```diff
@@ -266,6 +266,36 @@ impl FuzzProject {
         Ok(cmd)
     }
 
+    fn target_dir(&self, build: &options::BuildOptions) -> Result<PathBuf> {
+        if build.coverage {
+            Ok(env::current_dir()?
+                .join("target")
+                .join(default_target())
+                .join("coverage"))
+        } else {
+            Ok(build
+                .target_dir
+                .as_ref()
+                .map(PathBuf::from)
+                .unwrap_or_else(|| self.project_dir.join("target")))
```

Comment:
`self.project_dir.join("target")` is going to be incorrect if the user using either the `CARGO_TARGET_DIR=...` environment variable or configuring it via `.cargo/config.toml`. In this case, we should return `None`.